proposals/dark_mode_extension_icons.md

```diff
@@ -0,0 +1,231 @@
+# Dark Mode Extension Icons
+
+**Summary**
+
+Feature to enable developers to enhance extension icon visibility in dark mode.
+
+**Document Metadata**
+
+**Author:** solomonkinard
+
+**Sponsoring Browser:** Chromium
+
+**Contributors:** oliverdunk, xeenon, carlosjeurissen, hanguokai
+
+**Created:** 2024-04-05
+
+**Related Issues:**
+* https://crbug.com/893175
+* https://github.com/w3c/webextensions/issues/229
+
+## Motivation
+
+### Objective
+
+Extension developers will be able to supply and define a set of icons to be used
+in the event that user has expressed the preference for a page that has a dark
+theme.
+
+#### Use Cases
+
+1. Improved icon visibility in the extension toolbar.
+1. Improved icon visibility on the management and shortcuts pages.
+1. Dark mode icon declarations made possible through the extension manifest.
+1. `setIcon()` will allow setting of dark and/or light mode specific icons.
+1. Improved icon visibility on context menu (Chrome relies on default
+icons, only Firefox and Safari can specify menu icons).
+1. Improved icon visibility on side panel (Chrome relies on default icons,
+Firefox can specify icons).
+
+### Known Consumers
+
+The Chromium bug has a significant amount of developer interest.
+
+## Specification
+
+### Schema
+
+manifest.json
+```
+"icon_variants": [
+  {
+    "any": "any.svg",
+  },
+  {
+    "16": "16.png",
+    "32": "32.png",
+  },
+  {
+    "16": "dark16.png",
+    "32": "dark32.png",
+    "color_scheme": "dark",
+  },
+  {
+    "16": "light16.png",
+    "32": "light32.png",
+    "color_scheme": ["dark", "light"]
+  }
+]
+```
+  `icon_variants` requires an array with a set of icon groups objects. Each icon
+group consists of a set of icon paths and icon group matching criteria.
+
+  The icon paths are set using the size as key and/or the keyword `"any"`.
+
+  Optionally the `color_scheme` matching criterium which can either be an string or
+array of color schemes.
+
+Setting `icon_variants` dynamically:
+```
+const exampleProperties: {string: string | ImageData}[] = [
+  {
+    "any": "any.svg",
+  },
+  {
+    "16": "16.png",
+    "32": "32.png"
+  },
+  {
+    "16": darkImageData16,
+    "32": darkImageData32,
+    "color_scheme": "dark"
+  },
+  {
+    "16": lightImageData16,
+    "32": lightImageData32,
+    "color_scheme": "light"
+  }
+];
+
+// action.setIcon().
+const createProperties = {variants: exampleProperties};
+action.setIcon(createProperties);
+
+// menus.*(), for supporting browsers.
+const menusProperties = {icon_variants: exampleProperties};
+menus.create(menusProperties);
+menus.update(menusProperties);
```

Comment:
```suggestion
menus.update(id, menusProperties);
```

Note: menus is supported by Firefox and Safari; Chrome only supports contextMenus (but icons is not supported, yet? We just discussed this addition today). If you prefer, feel free to replace "icons" with "contextMenus" (which are also supported by Firefox and Safari as aliases of menus).